src/mlpack/methods/ann/loss_functions/KLDivergence_impl.hpp

```diff
@@ -0,0 +1,73 @@
+/**
+ * @file KLDivergence_impl.hpp
+ * @author Dakshit Agrawal
+ *
+ * Implementation of the KL Divergence error function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_ANN_LOSS_FUNCTION_KLDIVERGENCE_IMPL_HPP
+#define MLPACK_METHODS_ANN_LOSS_FUNCTION_KLDIVERGENCE_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "KLDivergence.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+KLDivergence<InputDataType, OutputDataType>::KLDivergence(const bool takeMean) :
+    takeMean(takeMean)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename TargetType>
+double KLDivergence<InputDataType, OutputDataType>::Forward(
+    const InputType&& input, const TargetType&& target)
+{
+  // Input and Target should be of same size.
+  double loss;
```

Comment:
No need to define the parameter here, we can return right away.

Reply:
Done :)